File: test/ch21-benchmarks/bench-003-string-concat.jl
# BENCH-003: String concatenation (10K operations) - Julia
function string_concatenation(iterations::Int64)::String
    # Idiomatic: repeat function (most efficient)
    return repeat("x", iterations)
end

function main()
    iterations = 10000
    result = string_concatenation(iterations)
    # Silent for benchmarking
    # Expected: 10000 characters
end

main()

File: test/ch21-benchmarks/bench-003-string-concat.go
// BENCH-003: String concatenation (10K operations) - Go
package main

import "strings"

func stringConcatenation(iterations int) string {
	// Idiomatic: strings.Builder (efficient, O(n))
	var builder strings.Builder
	for i := 0; i < iterations; i++ {
		builder.WriteString("x")
	}
	return builder.String()
}

func main() {
	iterations := 10000
	result := stringConcatenation(iterations)
	_ = result
	// Silent for benchmarking
	// Expected: 10000 characters
}
